### ISSUE TITLE
EXCEPTION_ACCESS_VIOLATION (0xc0000005) - After updating to Adoptium Java 21 in Apache Tomcat 9.x version, server is getting force shutdown with fatal error

### ISSUE THREAD

**Issue by ebadugu** [adoptium/adoptium-support#1117]
## Question
A fatal error has been detected by the Java Runtime Environment - While performing or running web application what could be the potential issue?

```
A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00007ffbc54e00cd, pid=33596, tid=21584
#
# JRE version: OpenJDK Runtime Environment Temurin-21.0.3+9 (21.0.3+9) (build 21.0.3+9-LTS)
# Java VM: OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (21.0.3+9-LTS, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, windows-amd64)
# Problematic frame:
# V  [jvm.dll+0x5d00cd]
```

# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00000261f3b1a8cf, pid=27908, tid=29584
#
# JRE version: OpenJDK Runtime Environment Temurin-21.0.3+9 (21.0.3+9) (build 21.0.3+9-LTS)
# Java VM: OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (21.0.3+9-LTS, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, windows-amd64)
# Problematic frame:
# j  javax.servlet.http.HttpServletResponseWrapper.addIntHeader(Ljava/lang/String;I)V+1
#
# No core dump will be written. Minidumps are not enabled by default on client versions of Windows

## Context

Java version:

Your operating system and platform:
Windows 10

**Comment by karianna** [comment]
@ebadugu can you post the full hs_err_<pid> crash log?

**Comment by ebadugu** [comment]
@karianna  - Thanks for your response.  PFA last 3 occurrences of hs_err_crash log files.

[hs_err_pid18748.log](https://github.com/user-attachments/files/15950259/hs_err_pid18748.log)

[hs_err_pid30136.log](https://github.com/user-attachments/files/15950262/hs_err_pid30136.log)
 
[hs_err_pid13184.log](https://github.com/user-attachments/files/15950263/hs_err_pid13184.log)

**Comment by ebadugu** [comment]
@karianna - Any pointers captured from the attached crash logs? Please let me know.

**Comment by karianna** [comment]
@ebadugu I took a look at all three logs and they're all crashing in the HttpServlet space as such (albeit on different methods each time).  Often that means you've got some sort of byte code weaving agent (or similar) that might be interfering.  Can -agentpath:C:\Program Files\SentinelOne\Sentinel Agent 23.4.4.223\SentinelJava64.dll=extendedCapabilities  be disabled and then see if the crashes stop?

Also, what exact version of Tomcat are you using?

**Comment by ebadugu** [comment]
@karianna Thanks for your pointers, I will verify the issue by disabling the agent you suggested. We are currently using the Apache Tomcat 9.0.89 version.

**Comment by BAGELreflex** [comment]
We had a customer experiencing this same issue occurring on Tomcat 9.0.87 and Adoptium Java 8.0.402+6. We also noticed that the customer had 2x SentinelOne dll files loaded via `-agentlib` being injected to the runtime command somehow. We had the customer whitelist the Tomcat installation directory in SentinelOne and restart Tomcat. They have not reported this issue since making that change. We have advised the customer to bring this to the attention of SentinelOne.
